plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -223,32 +159,27 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
              "the process");
       }
 
-      // avoid accumulating all these posts during replay before ship threads started
+      // avoid accumulating a bunch of post()s during replay before ship threads started
       // that can lead to a large memory consumption and failures
       // this is safe as there are no clients connected until after replay is complete
       // this method is called from the main thread and "plugin_started" is set on the main thread as well when plugin is started 
```

Comment:
Not only is the comment incorrect but it seems the whole `plugin_started` check can be removed: it's impossible for there to be anything in `connections` until the ship plugin is started (because that's where the listener is started), and the plugin won't be started until after replay completes.

Reply:
I've removed all the `plugin_started` stuff in 598e4b2 :tada: